python/feast/feast.py

```diff
@@ -0,0 +1,589 @@
+from ctypes import cdll, POINTER, c_uint, c_double
+import inspect
+from numpy import zeros
+import numpy as np
+from numpy.ctypeslib import ndpointer
+from logging import getLogger
+
+
+log = getLogger()
+
+# load the library
+fstoolbox_lib = cdll.LoadLibrary("libFSToolbox.so")
+
+
+def _load(config):
+    # Set the expected argtypes and return types for library functions
+    for func_name in config.func_names:
+        f = getattr(fstoolbox_lib, func_name, None)
+        if f is not None:
+            f.argtypes = config.argtypes
+            f.restype = config.restype
+        else:
+            print("Failed to set types for %s", func_name)
+
+class Config:
+    def __init__(self, func_names, argtypes, restype):
+        self.func_names = func_names
+        self.argtypes = argtypes
+        self.restype = restype
+
+# Declare useful return types that are a bit special (**)
+_uintpp = ndpointer(dtype=np.uintp, ndim=1, flags='F')
+_doublepp = ndpointer(dtype=np.uintp, ndim=1, flags='C')
+
+# Create the configurations to load
+normLibConfig = Config(
+    ["CMIM", "CondMI", "DISR", "ICAP", "JMI", "MIM", "mRMR_D"],
+    [c_uint, c_uint, c_uint, _uintpp, POINTER(c_uint), POINTER(c_uint), POINTER(c_double)],
+    POINTER(c_uint)
+)
+discLibConfig = Config(
+    ["discCMIM", "discCondMI", "discDISR", "discICAP", "discJMI", "discMIM", "disc_mRMR_D"],
+    [c_uint, c_uint, c_uint, _doublepp, POINTER(c_double), POINTER(c_double), POINTER(c_double)],
+    POINTER(c_double)
+)
+weightedLibConfig = Config(
+    ["weightedCMIM", "weightedCondMI", "weightedDISR", "weightedJMI", "weightedMIM"],
+    [c_uint, c_uint, c_uint, _uintpp, POINTER(c_uint), POINTER(c_double), POINTER(c_uint), POINTER(c_double)],
+    POINTER(c_uint)
+)
+discWeightedLibConfig = Config(
+    ["discWeightedCMIM", "discWeightedCondMI", "discWeightedDISR", "discWeightedJMI", "discWeightedMIM"],
+    [c_uint, c_uint, c_uint, _doublepp, POINTER(c_double), POINTER(c_double), POINTER(c_double), POINTER(c_double)],
+    POINTER(c_double)
+)
+betaGammaConfig = Config(
+    ["BetaGamma"],
+    [c_uint, c_uint, c_uint, _uintpp, POINTER(c_uint), POINTER(c_uint), POINTER(c_double), c_double, c_double],
+    POINTER(c_uint)
+)
+discBetaGammaConfig = Config(
+    ["discBetaGamma"],
+    [c_uint, c_uint, c_uint, _doublepp, POINTER(c_double), POINTER(c_double), POINTER(c_double), c_double, c_double],
+    POINTER(c_double)
+)
+
+# list and load all configs
+configs = [
+    normLibConfig, 
+    discLibConfig, 
+    weightedLibConfig,
+    discWeightedLibConfig, 
+    betaGammaConfig, 
+    discBetaGammaConfig
+]
+for config in configs:
+    _load(config)
+
+
+
+def FeatureSelectionCommon(data_set, labels, num_features_to_select, *args, **kwargs):
```

Comment:
Can this be made module private (i.e., start with an `_`)? I worry users might get weird error messages if they accidentally supply arguments which are unexpected for the underlying C function.

Reply:
oh absolutely I can make this one private. All of them will be made private and I can change the target as well as add some additional information to the setup. I will add a python readme too. I wrote a bash script that will allow the user to run everything from a single executable file to make the setup a bit easier.

Reply:
Thanks. For Python it would probably be nicer to figure out how to package the native libraries into the python library install as that's how most of the other Python libraries with native extensions work, but I can look at doing that separately.